cmake/Modules/CompilerConfig.cmake

```diff
@@ -5,6 +5,13 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+set(CMAKE_C_STANDARD 17)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+# TODO/FIXME: Investigate disabling C extensions on Linux
+if(OS_WINDOWS AND MSVC)
+  set(CMAKE_C_EXTENSIONS OFF)
+endif()
```

Comment:
To maintain similar functionality between the OBS_CMAKE 2 and 3 paths (until we retire OBS_CMAKE 2 as an option), shouldn't this be:
```
if((OS_WINDOWS AND MSVC) OR OS_MACOS)
  set(CMAKE_C_EXTENSIONS OFF)
endif()
```
or:
```
if(NOT OS_LINUX)
  set(CMAKE_C_EXTENSIONS OFF)
endif()
```

Reply:
If we still care about CMake 2.0 working, then yeah, the latter would be fine.